evm/contracts/Propcorn.sol

```diff
@@ -121,81 +153,68 @@ contract Propcorn is Initializable, UUPSUpgradeable, OwnableUpgradeable {
         if (feeBasisPoints > 10000) {
             revert InvalidFee();
         }
-        _proposals[msg.sender].push(
+
+        proposals.push(
             Proposal(
                 url,
                 secondsToUnlock,
                 0,
                 minAmountRequested,
                 0,
                 feeBasisPoints,
+                msg.sender,
                 ProposalStatus.FUNDING
             )
         );
 
         emit ProposalCreated(
             msg.sender,
-            _proposals[msg.sender].length - 1,
+            proposals.length - 1,
             url,
             secondsToUnlock,
             minAmountRequested,
             feeBasisPoints
         );
     }
 
-    function fundProposal(
-        address account,
-        uint256 index
-    ) public payable proposalExists(account, index) {
-        Proposal storage proposal = _proposals[account][index];
+    function fundProposal(uint256 index) public payable proposalExists(index) {
+        Proposal storage proposal = proposals[index];
 
         _checkFundability(proposal);
 
-        _addressAndProposalToBalance[
-            uint256(keccak256(abi.encodePacked(msg.sender, account, index)))
-        ] += msg.value;
+        funderToProposalBalance[msg.sender][index] += msg.value;
         proposal.balance += msg.value;
 
         if (proposal.balance >= proposal.minAmountRequested) {
             proposal.status = ProposalStatus.STARTED;
             proposal.fundingCompletedAt = block.timestamp;
         }
 
-        emit ProposalFunded(msg.sender, account, index, msg.value);
+        emit ProposalFunded(msg.sender, proposal.author, index, msg.value);
     }
 
-    function defundProposal(
-        address account,
-        uint256 index
-    ) public proposalExists(account, index) {
-        Proposal storage proposal = _proposals[account][index];
+    function defundProposal(uint256 index) public proposalExists(index) {
+        Proposal storage proposal = proposals[index];
 
         _checkFundability(proposal);
 
-        uint256 key = uint256(
-            keccak256(abi.encodePacked(msg.sender, account, index))
-        );
-
-        uint256 toReturn = _addressAndProposalToBalance[key];
+        uint256 toReturn = funderToProposalBalance[msg.sender][index];
```

Comment:
😬

Reply:
in the end that had to change as well, given that the author address is not part of the key anymore